core/src/main/scala/codesearch/core/index/SourceDownloader.scala

```diff
@@ -0,0 +1,43 @@
+package codesearch.core.index
+import java.io.File
+
+import ammonite.ops.pwd
+import sys.process._
+
+import scala.concurrent.Future
+
+case class TargetPackage(
+    name: String,
+    version: String
+)
+
+trait SourceDownloader {
```

Comment:
mb SourceRepository? in future, it will enable create multiple repository implementations - e.g. Java - mvn and jfrog
name SourceRepository more clearly imho

Reply:
This will be changed in the following commits.